components/inputs/input-checkbox.js

```diff
@@ -38,7 +40,8 @@ export const checkboxStyles = css`
 	}
 	input[type="checkbox"].d2l-input-checkbox:hover,
 	input[type="checkbox"].d2l-input-checkbox:focus,
-	input[type="checkbox"].d2l-input-checkbox.d2l-input-checkbox-focus {
+	input[type="checkbox"].d2l-input-checkbox.d2l-input-checkbox-focus,
+	:host(.d2l-hovering) input[type="checkbox"].d2l-input-checkbox {
```

Comment:
Yeah I agree.  It took me a while to figure out how this was working before... since indeed the `:hover` selector was being used while hovering the item without hovering the actual checkbox.